potential buffer overflow in dotnet/tools/common/manifest.c

We compile with all warnings enabled, and warnings as error 

For this line , https://github.com/bazelbuild/rules_dotnet/blob/66a708c64da9126c640b7ca62f03357740d06ea4/dotnet/tools/common/manifest.c#L438, we get an clang compilation error due to a warning

```
[2022-05-23T20:02:41.423Z] ERROR: /private/var/tmp/_bazel_admin/ad1b4e7593ac44fb49644fac721d0a3e/external/io_bazel_rules_dotnet/dotnet/tools/common/BUILD:5:11: Compiling dotnet/tools/common/manifest.c failed: (Aborted): wrapped_clang failed: error executing command external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g 'DEBUG_PREFIX_MAP_PWD=.' -iquote ... (remaining 29 arguments skipped)
[2022-05-23T20:02:41.423Z] 
[2022-05-23T20:02:41.423Z] Use --sandbox_debug to see verbose messages from the sandbox
[2022-05-23T20:02:41.423Z] external/io_bazel_rules_dotnet/dotnet/tools/common/manifest.c:438:34: error: the value of the size argument in 'strncat' is too large, might lead to a buffer overflow [-Werror,-Wstrncat-size]
[2022-05-23T20:02:41.423Z]     strncat(buffer, "/MANIFEST", sizeof(buffer)-strlen(buffer));
[2022-05-23T20:02:41.423Z]                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[2022-05-23T20:02:41.423Z] external/io_bazel_rules_dotnet/dotnet/tools/common/manifest.c:438:34: note: change the argument to be the free space in the destination buffer minus the terminating null byte
[2022-05-23T20:02:41.423Z]     strncat(buffer, "/MANIFEST", sizeof(buffer)-strlen(buffer));
[2022-05-23T20:02:41.423Z]                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[2022-05-23T20:02:41.423Z]                                  sizeof(buffer) - strlen(buffer) - 1
[2022-05-23T20:02:41.423Z] 1 error generated.
[2022-05-23T20:02:41.423Z] Error in child process '/usr/bin/xcrun'. 1
```

I think the compiler is right, please see https://godbolt.org/z/jjPP96Pzs for a short example

--- a/dotnet/tools/common/manifest.c
+++ b/dotnet/tools/common/manifest.c
@@ -122,7 +122,7 @@ const char *GetPathFromManifestEntries(const char *curPath, const char *prefix)
         else
             *p = '\0';
 
-        /* Check if directory is already present */  
+        /* Check if directory is already present */
         for(i = 0; i < dirsCount; ++i)
             if (strcmp(dirs[i], dir)==0)
                 break;
@@ -149,7 +149,7 @@ const char *GetPathFromManifestEntries(const char *curPath, const char *prefix)
         strncat(result, ":", bufferSize - strlen(result));
         #endif
         free(dirs[i]);
-    }    
+    }
     free(dirs);
 
     /* Add current PATH to the end */
@@ -435,7 +435,7 @@ const char *GetManifestPath()
         return buffer;
 
     getcwd(buffer, sizeof(buffer));
-    strncat(buffer, "/MANIFEST", sizeof(buffer)-strlen(buffer));
+    strncat(buffer, "/MANIFEST", sizeof(buffer)-strlen(buffer)-1);
     if (IsVerbose())
         printf("4. Checking MANIFEST %s\n", buffer);
     if (access(buffer, F_OK) != -1)

